funcx_endpoint/tests/unit/test_funcx_worker.py

```diff
@@ -0,0 +1,145 @@
+import pickle
```

Comment:
What's the reasoning behind moving this file out of `tests/funcx_endpoint/executors/high_throughput`? Is the idea to eventually move the other files in the original directory to the new `tests/unit` directory? Otherwise I'm disinclined to agree with this change, since it breaks the pattern that the rest of the tests follow.

Reply:
Let me start by saying that I don't want to argue too strongly in favor of this change. I think it's better to have unit tests segregated into a dedicated `unit/` directory, but it's not relevant to the rest of this work. I can put this file where the old test was if that's desirable.

I would like to write unit tests in `tests/unit/` and acceptance or integration tests in directories like `tests/functional/` or `tests/integration/`. My primary reason is that `tox -e py -- tests/unit/` should run very fast, and provide good diagnostic information on changes.

I also have never, personally, bought into the idea of a test layout in which tests are kept in a dir tree which mirrors the source. It's fine when you have `Foo` from `foo.py` and `Bar` from `bar.py` which you want to test independently. But what if `Foo` and `Bar` are mutually recursive classes which provide some functionality in tandem? Do tests for that go into `test_foo.py` or `test_bar.py`? You might think the answer is obvious, or that it doesn't matter, but this is why I don't like this test layout. It purports to answer the issue of where you put different tests, but I don't think it does that very well and completely.